v2/tests/git_initializer_test.go

```diff
@@ -179,46 +231,150 @@ func TestMain(t *testing.T) {
-
-			for {
-				select {
-				case logEntry, ok := <-logEntryCh:
-					if ok {
-						log.Println(logEntry.Message)
-					} else {
-						logEntryCh = nil
-					}
-				case err, ok := <-errCh:
-					if ok {
-						t.Fatalf("error from log stream: %s", err)
-					}
-					errCh = nil
-				case <-ctx.Done():
-					break
-				}
-
-				// log and err channels empty, let's check worker status
-				if logEntryCh == nil && errCh == nil {
-					for {
-						event, err := client.Core().Events().Get(ctx, e.ID)
-						require.NoError(t, err, "error getting event")
-
-						phase := event.Worker.Status.Phase
-						if phase.IsTerminal() {
-							require.Equal(t, tc.terminalPhase, phase, "worker's terminal phase does not match expected")
-							break
-						}
-					}
-					break
-				}
+			assertWorkerTerminalPhase(t, ctx, client, e, tc)
+			assertWorkerLogs(t, ctx, client, e, tc)
+
+			// We expect the vcs init container to fail for this test,
+			// therefore, assertions on jobs are not applicable
+			if tc.name == "GitHub - vcs failure" {
+				return
 			}
+
+			assertJobTerminalPhase(t, ctx, client, e, tc)
+			assertJobLogs(t, ctx, client, e, tc)
 		})
 	}
 }
+
+func assertWorkerTerminalPhase(
+	t *testing.T,
+	ctx context.Context,
+	client sdk.APIClient,
+	e core.Event,
+	tc testcase) {
+
+	for {
+		event, err := client.Core().Events().Get(ctx, e.ID)
+		require.NoError(t, err, "error getting event")
+
+		phase := event.Worker.Status.Phase
+		if phase.IsTerminal() {
+			require.Equal(t, tc.expectedWorkerPhase, phase, "worker's terminal phase does not match expected")
+			break
+		}
+	}
+}
+
+func assertWorkerLogs(
+	t *testing.T,
+	ctx context.Context,
+	client sdk.APIClient,
+	e core.Event,
+	tc testcase) {
+
+	// Check vcs container logs
+	selector := &core.LogsSelector{Container: "vcs"}
+	logs := captureLogs(t, ctx, client, e, selector)
+
+	if tc.expectedVCSLogsContains != "" {
+		require.Contains(t, logs, tc.expectedVCSLogsContains, "vcs init container logs do not match expected")
+	}
+
+	// We expect the vcs init container to fail for this test,
+	// therefore, assertions on worker logs are not applicable
+	if tc.name == "GitHub - vcs failure" {
+		return
+	}
+
+	// Check worker container logs
+	selector = &core.LogsSelector{}
+	logs = captureLogs(t, ctx, client, e, selector)
+
+	if tc.expectedWorkerLogsContains != "" {
+		require.Contains(t, logs, tc.expectedWorkerLogsContains, "worker's logs do not match expected")
+	} else {
+		// Look for default logs that we expect must exist
+		require.Contains(t, logs, "brigade-worker version", "worker's logs do not match expected")
+	}
+}
+
+func assertJobTerminalPhase(
+	t *testing.T,
+	ctx context.Context,
+	client sdk.APIClient,
+	e core.Event,
+	tc testcase) {
+
+	for {
+		event, err := client.Core().Events().Get(ctx, e.ID)
+		require.NoError(t, err, "error getting event")
+
+		jobs := event.Worker.Jobs
+		require.Equal(t, 1, len(jobs), "expected job count is not 1")
+
+		job := jobs["ls"]
```

Comment:
Maybe the job name can be passed in instead of being hardcoded here? As it stands now, `assertJobTerminalPhase ` isn't useful for hypothetical test cases that are constructed differently and do not have a job named `ls`.

Reply:
Now tracked via a const.